odoo/addons/base/models/ir_qweb.py

```diff
@@ -613,16 +665,145 @@ def _render(self, template, values=None, **options):
+                        render_template = template_functions[iterator_info['method']]
+                    elif iterator_info['options']:
+                        iterator_info['options'] = irQweb._compile(iterator_info['view_ref'])[2]
+
+                    # Apply a new scope if needed
+                    if iterator_info['scope']:
+                        if iterator_info['scope'] == 'root':
+                            values = root_values
+                        values = values.copy()
+                    iterator_info['values'] = values
+
+                    # Update values ​​with default values
+                    if iterator_info['default_values'] is not None:
+                        values.update(iterator_info['default_values'])
+
+                    # Create the iterator from the template
+                    iterator_info['iterator'] = render_template(irQweb, values)
+
+                values = iterator_info['values']
+                irQweb = iterator_info['irQweb']
+                dont_fetch_directive = irQweb.env.context.get('__qweb_dont_fetch_directive')
+
+                if len(stack) > 100:
+                    raise RecursionError('Qweb template infinity loop')  # noqa: TRY301
+
+                # traverse the iterator
+                for item in iterator_info['iterator']:
+                    if isinstance(item, str) or item[5] == dont_fetch_directive:
+                        yield item
+                    else:
+                        stack.append({
+                            'iterator': None,
+                            'values': None,
+                            'options': None,
+                            'irQweb': None,
+                            'context': item[0],
+                            'view_ref': item[1],
+                            'method': item[2],
+                            'default_values': item[3],
+                            'scope': item[4],
+                            'directive': item[5],
+                            'caller_path_xml': item[6],
+                        })
+                        break
+                else:
+                    stack.pop()
+
+            except (TransactionRollbackError, ReadOnlySqlTransaction):
+                raise
+            except QWebException as error:
+                error.path_info = OrderedSet(info['caller_path_xml'] for info in reversed(stack) if info['caller_path_xml'])
+                if error.ref is None:
+                    error.ref = iterator_info['view_ref']
+                raise
+            except Exception as error:
+                loaded_codes = self.env.context['__qweb_loaded_codes']
+                if (iterator_info['view_ref'] in loaded_codes and not isinstance(error, RecursionError)) or len(stack) <= 1:
+                    options = iterator_info['options']
+                    ref = options.get('ref') if options else view_ref
+                    ref_name = options.get('ref_name') if options else None
```

Comment:
Maybe the following (not strictly equivalent to view_ref).
```suggestion
                    options = iterator_info['options'] or {}
                    ref = options.get('ref', view_ref)
                    ref_name = options.get('ref_name')
```